src/components/web-monitoring-ui.jsx

```diff
@@ -65,14 +68,29 @@ export default class WebMonitoringUi extends React.Component {
   logOut () {
     api.logOut();
     this.setState({user: api.userData});
-    this.loadPages('pages');
+    this.loadPages();
   }
 
-  search (query) {
-    this.setState({search: query});
+  urlSearch (urlQuery) {
+    this.setState({urlQuery});
     this.loadPages();
   }
 
+  dateSearch(startDateQuery, endDateQuery) {
+    this.setState({startDateQuery, endDateQuery}, function () {
+      let captureTime = null;
+      if (this.state.startDateQuery || this.state.endDateQuery) {
+        captureTime = {
+          capture_time: {
+            startDateQuery: this.state.startDateQuery,
+            endDateQuery: this.state.endDateQuery
+          }
+        };
+      }
```

Comment:
This formatting should really happen either before we set the state or inside `loadPages()` when we are formatting the arguments for `api.getPages()`. (Depends on whether you want to view the state as “current query args” or as “the data we have set in various fields.”)